### ISSUE TITLE
[Feature request] See track history

### ISSUE THREAD

**Issue by TimSC** [owntracks/android#1152]
A fairly standard feature of tracking apps is to see the track history of a single user. I know [recorder](https://github.com/owntracks/recorder) provides something like this but we could do with it in the mobile app as well.

**Comment by growse** [comment]
I've thought about this before. The main challenge is now the app needs to store a history of points, so coming up with a sensible value for "how many points?", as well as other things like "do we also keep the inaccurate locations?" and "what about points for contacts?".

None of that is particularly hard though, we can aim to have something simple in the next major release and see what people think.

**Comment by TimSC** [comment]
Just to clarify, I was thinking of the track history for a friend, as well as yourself. This would require changes to the server side?

**Comment by ckrey** [comment]
in iOS we keep the last `positions` (setting) positions of the user and for the contacts. Default ist 50. „Inaccurate positions“ are not sent and not stored. We show the track on the map when selected. There is no list view of the track and no export of track points in the app.

**Comment by growse** [comment]
> Just to clarify, I was thinking of the track history for a friend, as well as yourself. This would require changes to the server side?

The app can store a list of positions it receives for a contact, so that ask shouldn't require any changes outside of the app.

**Comment by TimSC** [comment]
So there would be no history while the phone is off or out of network connectivity?

**Comment by growse** [comment]
Yes, without a live connection to an instance, the client's not going to receive any locations from contacts. It can obviously store its own location history while offline, but there's no mechanism in the OT protocol / spec to allow a client to pull a history of locations for a contact from a recorder instance.

**Comment by ckrey** [comment]
If you are using an MQTT backend and use QoS>0 which is the default, messages from contacts will be queued until you are back online.

**Comment by compuguy** [comment]
Maybe the easiest way to implement this would be to allow a config option to point to a OwnTracks Frontend page? This doesn't work if the phone is offline, but its a thought.

**Comment by jpmens** [comment]
Our OwnTracks iOS app in combination with the Recorder can do something we call [Tours](https://owntracks.org/booklet/features/tours/).

**Comment by growse** [comment]
> Maybe the easiest way to implement this would be to allow a config option to point to a OwnTracks Frontend page? This doesn't work if the phone is offline, but its a thought.

I think the idea of being able to easily link to a recorder instance is reasonable, but the challenge is: how does the app know where the recorder is?

It might be guessable on HTTP mode, but not MQTT, it's decoupled from the recorder completely.

Tours is coming, but it's not quite the same use case.

Still aiming to do simple history for self, but had to jettison from 2.5 in order to get the thing out of the door this decade.

**Comment by ckrey** [comment]
The app knows where the recorder is by requesting a new tour or the list of tours from the recorder. The recorder responds with the details including the tour's URL
